core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/MetricsServiceImpl.java

```diff
@@ -490,6 +490,22 @@ void setDataAccess(DataAccess dataAccess) {
     }
 
     @Override
+    public <T> Observable<T> findGaugeData(String tenantId, MetricId id, Long start, Long end,
+            Func1<Observable<DataPoint<Double>>, Observable<T>>... funcs) {
+
+        Observable<DataPoint<Double>> dataCache = findGaugeData(tenantId, id, start, end).cache();
+        Observable<T> result = null;
+
+        for (Func1<Observable<DataPoint<Double>>, Observable<T>> func : funcs) {
+            result = (null == result) ?
+                    func.call(dataCache) :
+                    result.concatWith(func.call(dataCache));
+        }
+
+        return result;
+    }
```

Comment:
This method can be written more simply as,

```java
@Override
public <T> Observable<T> findGaugeData(String tenantId, MetricId id, Long start, Long end,
    Func1<Observable<DataPoint<Double>>, Observable<T>>... funcs) {

    Observable<DataPoint<Double>> dataCache = findGaugeData(tenantId, id, start, end).cache();
    return Observable.from(funcs).flatMap(fn -> fn.call(dataCache));
}
```
Aside from making the code more tidy, there is another good reason for the change. We can easily parallelize this with,

```java
return Observable.from(funcs).flatMap(fn -> fn.call(dataCache)).subscribeOn(Schedulers.compute());
```
The convention with a lot of the RxJava APIs is two have two versions of a method, one of which takes a Scheduler.